python/cudf/cudf/core/column/column.py

```diff
@@ -1207,6 +1207,11 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             arbitrary = cupy.asarray(arbitrary).astype(arb_dtype)
```

Comment:
If arbitrary is masked this won't work.

Reply:
Resolved this issue by separating out mask before we create a cupy array.